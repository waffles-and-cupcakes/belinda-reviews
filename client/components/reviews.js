import React from 'react';
import ReactDOM from 'react-dom';
import axios from 'axios';

class Reviews extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      total: 5,
      reviews: []
    }
  }

  componentDidMount() {
    axios.get('/rooms/1/reviews')
    .then((res) => {
      console.log('RESPONSE: ', res);
      this.setState({
        reviews: res.data
      });
    })
    .catch((err) => {
      console.log('error GET-ing /rooms/1/reviews')
    })
  }

  render() {
    return (
      <div> {JSON.stringify(this.state.reviews)}</div>
    );
  }
}

export default Reviews;