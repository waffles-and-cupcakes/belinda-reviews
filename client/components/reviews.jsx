import React from 'react';
import ReactDOM from 'react-dom';


class Reviews extends React.Component {

  render() {
    return (
      <section className="review-container"> 

        <div className="detail-container">
          <img className="user-photo" src="./pfp.jpg" />
          <div className="x-detail-container">
            <p> Kristen </p>
            <p> June 2017 </p>
          </div>
        </div>

        <div className="review-body"> 
          A beautiful space! We loved all the attention to 
          detail and local recommendations. Would love to se all
            their properties!
        </div>

      </section> 
    )
    
  }
}

export default Reviews;