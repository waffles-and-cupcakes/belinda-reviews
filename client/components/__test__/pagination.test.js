import React from 'react';
import renderer from 'react-test-renderer';
import Pagination from '../pagination';
import reviews from '../../mockData';

test('Pagination renders', () => {
  const component = renderer.create(<Pagination
    numPages={Math.ceil(reviews.length / 7)}
    changePage={jest.fn()}
    activePage="1"
  />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Pagination changes page on handlePageClick', () => {
  const component = renderer.create(<Pagination
    numPages={Math.ceil(reviews.length / 7)}
    changePage={jest.fn()}
    activePage="1"
  />);
  const tree = component.toJSON();

  tree.children[0].props.onClick({ target: { textContent: '3' } });

  expect(tree).toMatchSnapshot();
});

