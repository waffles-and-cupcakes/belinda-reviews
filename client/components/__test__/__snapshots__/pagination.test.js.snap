// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pagination changes page on handlePageClick 1`] = `
<ul
  className="row pagination pagination"
>
  <li
    className="pg-active pg-num"
    onClick={[Function]}
  >
    <a
      href="#!"
    >
      1
    </a>
  </li>
  <li
    className="pg-num"
    onClick={[Function]}
  >
    <a
      href="#!"
    >
      2
    </a>
  </li>
  <li
    className="pg-num"
    onClick={[Function]}
  >
    <a
      href="#!"
    >
      3
    </a>
  </li>
  <li
    className="pg-num"
    onClick={[Function]}
  >
    <a
      href="#!"
    >
      4
    </a>
  </li>
</ul>
`;

exports[`Pagination renders 1`] = `
<ul
  className="row pagination pagination"
>
  <li
    className="pg-active pg-num"
    onClick={[Function]}
  >
    <a
      href="#!"
    >
      1
    </a>
  </li>
  <li
    className="pg-num"
    onClick={[Function]}
  >
    <a
      href="#!"
    >
      2
    </a>
  </li>
  <li
    className="pg-num"
    onClick={[Function]}
  >
    <a
      href="#!"
    >
      3
    </a>
  </li>
  <li
    className="pg-num"
    onClick={[Function]}
  >
    <a
      href="#!"
    >
      4
    </a>
  </li>
</ul>
`;
